container_api/src/config.rs

```diff
@@ -0,0 +1,223 @@
+use holochain_core_types::{
+    entry::agent::{Agent, Identity},
+    error::{HcResult, HolochainError},
+    json::JsonString,
+};
+use holochain_dna::Dna;
+use serde::Deserialize;
+use std::{convert::TryFrom, fs::File, io::prelude::*};
+
+#[derive(Deserialize)]
+pub struct Configuration {
+    #[serde(rename = "agent")]
+    agents: Option<Vec<AgentConfiguration>>,
+    #[serde(rename = "dna")]
+    dnas: Option<Vec<DNAConfiguration>>,
+    #[serde(rename = "instance")]
+    instances: Option<Vec<InstanceConfiguration>>,
+    #[serde(rename = "interface")]
+    interfaces: Option<Vec<InterfaceConfiguration>>,
+    #[serde(rename = "bridge")]
+    bridges: Option<Vec<Bridge>>,
```

Comment:
Might be more idiomatic to just use the plural names in the toml config too, i.e. `[[bridges]]`. The [toml docs](https://github.com/toml-lang/toml#array-of-tables) suggest this.

Reply:
Had it that way first. @sphinxc0re convinced me to do this way to have a more natural language compatible read of config files. I don't have a strong opinion but lean towards singular. Anybody with a strong opinion and good argument for it?